pkg/controllers/jobset_controller.go

```diff
@@ -219,6 +222,63 @@ func (r *JobSetReconciler) getChildJobs(ctx context.Context, js *jobset.JobSet)
 	return &ownedJobs, nil
 }
 
+func (r *JobSetReconciler) calculateAndUpdateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Prepare replicatedJobsReady for optimal iteration
+	replicatedJobsReady := map[string]int32{}
+	for _, replicatedJob := range js.Spec.ReplicatedJobs {
+		replicatedJobsReady[replicatedJob.Name] = 0
+	}
+
+	// Calculate ReadyPods for each Replicated Job
+	for _, job := range append(jobs.active, append(jobs.successful, jobs.failed...)...) {
+		ready := pointer.Int32Deref(job.Status.Ready, 0)
+		// parallelism is always set as it is otherwise defaulted by k8s to 1
+		podsCount := *(job.Spec.Parallelism)
+		if job.Spec.Completions != nil && *job.Spec.Completions < podsCount {
+			podsCount = *job.Spec.Completions
+		}
+		if job.Status.Succeeded+ready >= podsCount {
+			if job.Labels != nil && job.Labels[jobset.ReplicatedJobNameKey] != "" {
+				replicatedJobsReady[job.Labels[jobset.ReplicatedJobNameKey]]++
+			} else {
+				log.Error(nil, fmt.Sprintf("job %s missing ReplicatedJobName label", job.Name))
+			}
+		}
+	}
+
+	status := js.Status.DeepCopy().ReplicatedJobsStatus
+	// Update current ReplicatedJobsStatuses
+	for index := range status {
+		val, ok := replicatedJobsReady[status[index].Name]
+		if !ok {
+			log.Info(fmt.Sprintf("the %s replicated job is not anymore relevant", status[index].Name))
+			status = append(status[:index], status[index+1:]...)
+		} else {
+			status[index].ReadyJobs = val
+		}
+		delete(replicatedJobsReady, status[index].Name)
+	}
+
+	// Add new ReplicatedJobsStatuses
+	for name, readyJobs := range replicatedJobsReady {
+		status = append(status, jobset.ReplicatedJobStatus{
```

Comment:
If we only ever append to status, won't it just grow infinitely? Shouldn't we be updating each ReplicatedJobStatus in place?

Reply:
On line 251 I'v created a variable to keep all ReplicatedJobStatuses positions using current status DeepCopy, I update them, and on 266 I append the rest.
I use this mode so that the positions stay the same and it is smooth to read.